src/Renci.SshNet/Security/Cryptography/Ciphers/AesCipher.cs

```diff
@@ -1,86 +1,114 @@
-﻿using System;
+using System;
 using System.Security.Cryptography;
 
+using Renci.SshNet.Security.Cryptography.Ciphers.Modes;
+using Renci.SshNet.Security.Cryptography.Ciphers.Paddings;
+
 namespace Renci.SshNet.Security.Cryptography.Ciphers
 {
+    /// <summary>
+    /// Custom AES Cipher Mode, follows System.Security.Cryptography.CipherMode.
+    /// </summary>
+    public enum AesCipherMode
```

Comment:
We should name this something without "Aes", but I don't have any great suggestions.

1. `CipherModeType` / `CipherType` ?
2. Nest it in `Cipher` / `BlockCipher` / `CipherMode` as `Mode` / `Type` ? e.g.
```diff
namespace Renci.SshNet.Security.Cryptography;

public class Cipher
{
+   public enum Mode
+   {
+       CBC = 1,
    // etc
+   }
}
```
Then we have `new AesCipher(key, iv, Cipher.Mode.CBC);`

3. Remove it, replace it with `S.S.C.CipherMode` and define a constant value on `CipherMode` e.g.
```diff
namespace Renci.SshNet.Security.Cryptography.Ciphers;

public abstract class CipherMode
{
+   public const System.Security.Cryptography.CipherMode CTR = (System.Security.Cryptography.CipherMode)(-1);
}
```
Then we have `new AesCipher(key, iv, S.S.C.CipherMode.CBC)` and `new AesCipher(key, iv, Renci..CipherMode.CTR)`

I'm not hot on any of them.

Reply:
It's not ideal, but I think it's OK as this is only used with AesCipher.cs.

Reply:
Right now it's only being used for the AES implementation, but a block cipher mode is not something specific to AES. Maybe we will use it for 3DES?

 We can let Wojtek be the arbiter.

Reply:
I'd keep **AesCipherMode**, and rename it if/when we decide to use it for other ciphers.
Breaking changes on this level shouldn't hurt many users.